### DISCUSSION TITLE
error while building for android rustdeskflutter/ndk_arm64.sh Please help

### DISCUSSION THREAD

**Discussion by Ahsan445** [rustdesk/rustdesk#551] [Q&A] [answered]
I am able to do cargo run successfully,
but when building for android i am getting this error even though i have download and added vcpkg, openssl, android ndk to path carefully and checked many times for issue in path.

Any help would be appreciated.
Thanks

below is the output when running ndk_arm_64.sh from restdesk dir.

root@debian:~/rustdesk# OPENSSL_DIR=$HOME/openssl AARCH64_LINUX_ANDROID_OPENSSL_LIB_DIR=$HOME/openssl VCPKG_ROOT=$HOME/vcpkg ANDROID_NDK_HOME=$HOME/android-ndk-r22b flutter/ndk_arm64.sh
[2022-05-18T10:27:11Z INFO  cargo_ndk::cli] Using NDK at path: /root/android-ndk-r22b
[2022-05-18T10:27:11Z INFO  cargo_ndk::cli] NDK API level: 21
[2022-05-18T10:27:11Z INFO  cargo_ndk::cli] Building targets: arm64-v8a
[2022-05-18T10:27:11Z INFO  cargo_ndk::cli] Building arm64-v8a (aarch64-linux-android)
   Compiling scrap v0.5.0 (/root/rustdesk/libs/scrap)
   Compiling magnum-opus v0.4.0 (https://github.com/open-trade/magnum-opus#3c3d0b86)
error: failed to run custom build command for `magnum-opus v0.4.0 (https://github.com/open-trade/magnum-opus#3c3d0b86)`

Caused by:
  process didn't exit successfully: `/root/rustdesk/target/release/build/magnum-opus-5f167d02c15f1160/build-script-build` (exit status: 101)
  --- stdout
  cargo:info=arm64-android
  cargo:rustc-link-lib=static=opus
  cargo:rustc-link-search=/root/vcpkg/installed/arm64-android/lib
  cargo:include=/root/vcpkg/installed/arm64-android/include
  rerun-if-changed=/root/.cargo/git/checkouts/magnum-opus-7580b098b18e2bc5/3c3d0b8/opus_ffi.h
  rerun-if-changed=/root/vcpkg/installed/arm64-android/include

  --- stderr
  /usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found
  /usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found, err: true
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', /root/.cargo/git/checkouts/magnum-opus-7580b098b18e2bc5/3c3d0b8/build.rs:66:18
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
[2022-05-18T10:27:14Z INFO  cargo_ndk::cli] If the build failed due to a missing target, you can run this command:
[2022-05-18T10:27:14Z INFO  cargo_ndk::cli] 
[2022-05-18T10:27:14Z INFO  cargo_ndk::cli]     rustup target install aarch64-linux-android
root@debian:~/rustdesk#

**Comment by rustdesk** [comment]
I tested on Ubuntu 18

**Reply by Ahsan445** [reply]
I am on Debian 11 could that be a problem?

**Comment by rustdesk** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/issues/565
